### PR TITLE
UTest 2

### PR DESCRIPTION
**Pull request by RealyUniqueName**
- drop support for non-`ITest` test cases
- `UTEST_IGNORE_DEPENDS` (see README.md)
- added test case name filter to Runner.addCases
- support `@:ignore` meta (old `@Ignored` meta is still supported too)
- support advanced minification for js target

### CODE REVIEW

File: src/utest/ui/Report.hx [hunk 1]
```diff
@@ -12,20 +12,20 @@ import neko.Web;
```
Comment (Simn) [inline src/utest/ui/Report.hx]: There's some code above which also has to go:

```haxe
#if php
import php.Web;
#elseif neko
import neko.Web;
#end
```

Comment (RealyUniqueName) [reply src/utest/ui/Report.hx]: Done

Comment (Simn) [reply src/utest/ui/Report.hx]: Thanks! There's another neko one in ui.text.PrintReport line 22.

**Comment by Simn** [comment]
I'm trying to use this branch over on https://github.com/HaxeFoundation/haxe/pull/11386, but at the moment everything is failing. It wasn't that bad a few days ago, so a recent commit must have broken quite a few things.

Some tests will require fixing because the assert functions are now type-safe, but there must be other problems going on.

**Comment by RealyUniqueName** [comment]
I'll try to check that Haxe branch in the nearest days.

**Pull request merged**